cli/azd/internal/appdetect/maven.go

```diff
@@ -0,0 +1,206 @@
+	if err != nil {
+		return "", err
+	}
+	err = unzip(wrapperPath, mavenDir)
+	if err != nil {
+		return "", fmt.Errorf("failed to unzip maven bin.zip: %w", err)
+	}
+	return mavenCommand, nil
+}
+
+func getAzdMvnDir() (string, error) {
+	azdMvnFolderName := "azd-maven"
+	userHome, err := os.UserHomeDir()
+	if err != nil {
+		return "", fmt.Errorf("unable to get user home directory: %w", err)
+	}
+	return filepath.Join(userHome, azdMvnFolderName), nil
+}
+
+func getAzdMvnCommand(mavenVersion string) (string, error) {
+	mavenDir, err := getAzdMvnDir()
+	if err != nil {
+		return "", err
+	}
+	azdMvnCommand := filepath.Join(mavenDir, "apache-maven-"+mavenVersion, "bin", "mvn")
+	return azdMvnCommand, nil
+}
+
+func downloadMaven(filepath string) error {
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer func(out *os.File) {
+		err := out.Close()
+		if err != nil {
+			log.Println("failed to close file. %w", err)
+		}
+	}(out)
+
+	resp, err := http.Get(mavenURL)
+	if err != nil {
+		return err
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Println("failed to close ReadCloser. %w", err)
+		}
+	}(resp.Body)
+
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+func unzip(src string, destinationFolder string) error {
+	reader, err := zip.OpenReader(src)
+	if err != nil {
+		return err
+	}
+	defer func(reader *zip.ReadCloser) {
+		err := reader.Close()
+		if err != nil {
+			log.Println("failed to close ReadCloser. %w", err)
+		}
+	}(reader)
+
+	for _, file := range reader.File {
+		destinationPath, err := getValidDestPath(destinationFolder, file.Name)
+		if err != nil {
+			return err
+		}
+		if file.FileInfo().IsDir() {
+			err := os.MkdirAll(destinationPath, os.ModePerm)
+			if err != nil {
+				return err
+			}
+		} else {
+			if err = os.MkdirAll(filepath.Dir(destinationPath), os.ModePerm); err != nil {
+				return err
+			}
+
+			outFile, err := os.OpenFile(destinationPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, file.Mode())
+			if err != nil {
+				return err
+			}
+			defer func(outFile *os.File) {
+				err := outFile.Close()
+				if err != nil {
+					log.Println("failed to close file. %w", err)
+				}
+			}(outFile)
+
+			rc, err := file.Open()
+			if err != nil {
+				return err
+			}
+			defer func(rc io.ReadCloser) {
+				err := rc.Close()
+				if err != nil {
+					log.Println("failed to close file. %w", err)
+				}
+			}(rc)
+
+			for {
+				_, err = io.CopyN(outFile, rc, 1_000_000)
+				if err != nil {
+					if errors.Is(err, io.EOF) {
+						break
+					}
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getValidDestPath(destinationFolder string, fileName string) (string, error) {
+	destinationPath := filepath.Clean(filepath.Join(destinationFolder, fileName))
+	if !strings.HasPrefix(destinationPath, destinationFolder+string(os.PathSeparator)) {
+		return "", fmt.Errorf("%s: illegal file path", fileName)
+	}
+	return destinationPath, nil
+}
```

Comment:
This func is used to solve this proglem:
https://github.com/azure-javaee/azure-dev/actions/runs/12353956134/job/34474195291
> ![image](https://github.com/user-attachments/assets/2079eb3b-57f5-4a95-883f-97d15d5adfb4)